File: 프로그래머스/unrated/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
class Solution {
    public String[] solution(String my_str, int n) {
       int len = my_str.length();
        String[] answer = new String[(len + n - 1) / n];
        int i = 0;
        int startIndex = 0;

        while (startIndex < len) {
            int endIndex = Math.min(startIndex + n, len);
            answer[i] = my_str.substring(startIndex, endIndex);
            startIndex += n;
            i++;
        }

        return answer;
    }
}

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.py
def solution(my_str, n):
    answer = []
    cnt = 0
    start = 0
    end = n
    while(cnt<len(my_str)/n):
        cnt += 1
        answer.append(my_str[start:end])
        start +=n 
        end += n
    return answer